tests/integration/credset_test.go

```diff
@@ -0,0 +1,516 @@
+//go:build integration
+
+package integration_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"get.porter.sh/operator/controllers"
+	"github.com/carolynvs/magex/shx"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	cl "k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	porterv1 "get.porter.sh/operator/api/v1"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("CredentialSet create", func() {
+	Context("when a new CredentialSet resource is created with secret source", func() {
+		It("should run porter", func() {
+			By("creating an agent action", func() {
+				ctx := context.Background()
+				ns := createTestNamespace(ctx)
+				name := "test-cs-" + ns
+				testSecret := "foo"
+				createTestSecret(ctx, name, testSecret, ns)
+				Log(fmt.Sprintf("create credential set '%s' for credset", name))
+				cs := NewTestCredSet(name)
+				cs.ObjectMeta.Namespace = ns
+				cred := porterv1.Credential{
+					Name: "insecureValue",
+					Source: porterv1.CredentialSource{
+						Secret: name,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, cred)
+				cs.Spec.Namespace = ns
+
+				Expect(k8sClient.Create(ctx, cs)).Should(Succeed())
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential set to apply")).Should(Succeed())
+				validateCredSetConditions(cs)
+
+				Log("install porter-test-me bundle with new credset")
+				inst := NewTestInstallation("cs-with-secret")
+				inst.ObjectMeta.Namespace = ns
+				inst.Spec.Namespace = ns
+				inst.Spec.CredentialSets = append(inst.Spec.CredentialSets, name)
+				inst.Spec.SchemaVersion = "1.0.0"
+				Expect(k8sClient.Create(ctx, inst)).Should(Succeed())
+				Expect(waitForPorter(ctx, inst, "waiting for porter-test-me to install")).Should(Succeed())
+				validateInstallationConditions(inst)
+
+				// Validate that the correct credential set was used by the installation
+				jsonOut := runAgentAction(ctx, "show-outputs", ns, []string{"installation", "outputs", "list", "-n", ns, "-i", inst.Spec.Name, "-o", "json"})
+				credsValue := gjson.Get(jsonOut, `#(name=="outInsecureValue").value`).String()
+				Expect(credsValue).To(Equal(testSecret))
+			})
+		})
+	})
+})
+var _ = Describe("CredentialSet secret does not exist", func() {
```

Comment:
Do you know what the overhead/implications are of splitting these up into separate describe sections vs multiple checks in the same one? I'm wondering if we setup/tearndown just the test namespace for each of these, or if it also redoes the envtest cluster too?